### PR TITLE
More movement tests

### PR DESCRIPTION
**Pull request by EranDaniel98**

### CODE REVIEW

File: ChessMovementTest/KingTest.cs [hunk 1]
```diff
@@ -0,0 +1,71 @@
...
+        public void TestBlackKing()
+        {
+            Board board = TestHelper.CreateTwoPlayerBoard();
+
+            List<ChessPiece> secondPieces = board.chessPiecesByColor[PlayerColor.GetColor(PlayerNumber.SecondPlayer)];
+            ChessPiece wKing = secondPieces[(int)SecondPiecesNumber.King];
+
+            TestHelper.ValidateMovementResults(wKing, "");
+
+            //black king
```
Comment (NitsanLitov) [inline ChessMovementTest/KingTest.cs:53]: delete comment

File: ChessMovementTest/KingTest.cs [hunk 2]
```diff
@@ -0,0 +1,71 @@
...
+            TestHelper.PrintAll(board);
+            TestHelper.ValidateMovementResults(wKing, "b3 d2 b4 c4 d4");
+            
+            board.Move(('c', 3), ('b', 4));
+            TestHelper.PrintAll(board);
+            TestHelper.ValidateMovementResults(wKing, "a3 b3 c3 a4 c4 a5 b5 c5");
+        }
+
+        [TestMethod]
+        public void TestBlackKing()
```
Comment (NitsanLitov) [inline ChessMovementTest/KingTest.cs:44]: you didnt check at least on one of the king colors that king can take a different piece

File: ChessMovementTest/KingTest.cs (hunk 1 above)
Comment (EranDaniel98) [reply ChessMovementTest/KingTest.cs:53]: Done!

File: ChessMovementTest/KingTest.cs (hunk 2 above)
Comment (EranDaniel98) [reply ChessMovementTest/KingTest.cs:44]: added a check for taking a piece!

**Pull request merged**